### PR TITLE
feat: add log fields for API call logging

### LINKED ISSUE #424: Log relevant information, e.g. "project" (i.e. owner/repo/tenant) in API calls
For the moment the format is:
```
OneDrive API [tenant/owner/repo] method url
```
Questions:
- should the response status also be logged?
- should we support passing multiple fields to the `OneDrive` constructor, where one could pass the high level operation, e.g. `preview /document`?

### PR DESCRIPTION
**Pull request by dominique-pfister**
fix #424

### CODE REVIEW

**Comment by dominique-pfister** [comment]
Log line would look like this:
```
OneDrive API [test-tenantid/ovner/repo]: PUT /drives/1/items/2/content?@microsoft.graph.conflictBehavior=replace
```

File: src/OneDrive.js [hunk 1]
```diff
@@ -127,12 +127,18 @@ export class OneDrive {
     opts.headers.authorization = `Bearer ${accessToken}`;
     const url = `https://graph.microsoft.com/v1.0${relUrl}`;
     try {
+      const { log, auth: { label, tenant } } = this;
+      if (label) {
```
Comment (tripodsan) [inline src/OneDrive.js:131]: I would log it in any case, so we can find missing instrumentations.

File: src/OneDrive.js [hunk 2]
```diff
@@ -125,14 +125,19 @@ export class OneDrive {
...
+
+    const { log, auth: { logFields, tenant } } = this;
     const url = `https://graph.microsoft.com/v1.0${relUrl}`;
+    const method = opts.method || 'GET';
+
     try {
       const { fetch } = this.fetchContext;
       const resp = await fetch(url, opts);
-      const rateLimit = RateLimit.fromHeaders(resp.headers);
+      log.info(`OneDrive API [tenant:${tenant}] ${logFields}: ${method} ${relUrl} ${resp.status}`);
```
Comment (dominique-pfister) [inline src/OneDrive.js:136]: change to log fields and status

File: test/onedrive.test.js [hunk 3]
```diff
@@ -481,7 +481,7 @@ describe('OneDrive Tests', () => {
       });
 
     const od = new OneDrive({
-      auth: DEFAULT_AUTH(),
+      auth: DEFAULT_AUTH({ logFields: { project: 'owner/repo' } }),
```
Comment (dominique-pfister) [inline test/onedrive.test.js:484]: log message now looks like:
```
OneDrive API [tenant:test-tenantid] [project:owner/repo]: GET /shares/u!aHR0cHM6Ly9vbmVkcml2ZS5jb20vYS9iL2MvZDA/driveItem 200
```

**Review by tripodsan — APPROVED** [review]

**Pull request merged**